homeassistant/components/netatmo/sensor.py

```diff
@@ -113,7 +113,7 @@ class NetatmoSensorEntityDescription(SensorEntityDescription, NetatmoRequiredKey
         name="Pressure",
         netatmo_name="pressure",
         entity_registry_enabled_default=True,
-        native_unit_of_measurement=UnitOfPressure.MBAR,
+        native_unit_of_measurement=UnitOfPressure.HPA,
```

Comment:
This is a breaking change.

Is that needed? As in, if you like prefer hPa, you can change the unit for your entity to your liking.

Reply:
I think the `device_class` should be adjusted to use `SensorDeviceClass.ATMOSPHERIC_PRESSURE`

Reply:
> This is a breaking change.

Because there is a device_class and the unit is compatible, it shouldn't be a breaking change.
However, I think adjusting the device_class is better it if always refers to atmospheric pressure.

Reply:
> This is a breaking change.
> 
> Is that needed? As in, if you like prefer hPa, you can change the unit for your entity to your liking.

The default should be hPa, as it is the correct unit for atmospheric pressure.

Reply:
? That is what the description already said. IMHO not enough to justify a breaking change. See also comment from epenwt above.

Reply:
Changing the `device_class` is NOT a breaking change (existing sensors will keep using the old unit, new sensors will get the new unit).

Regarding the `native_unit_of_measurement`, the netatmo documentation specifies `mbar` in https://dev.netatmo.com/apidocumentation/weather so I think it's better to keep it as it is and avoid any chance of a breaking change.

Reply:
@frenck, @epenet this should not be a breaking change: thanks to https://github.com/home-assistant/core/pull/83228 a new unit, which is compatible with the old unit stored in the entity registry will be converted to the old unit.

Reply:
Cool! Still, see the last comment by @epenet, it is communicated as `mbar` by the API itself as well.

Reply:
Note: whatever you set it will be converted to hPa for "atmospheric_pressure"